tests/FSharp.Compiler.ComponentTests/FSharpChecker/TransparentCompiler.fs

```diff
@@ -894,4 +895,99 @@ let ``LoadClosure for script is recomputed after changes`` () =
         |> Seq.map (fun (k, g) -> k, g |> Seq.map fst |> Seq.toList)
         |> Map
 
-    Assert.Equal<JobEvent list>([Weakened; Requested; Started; Finished; Weakened; Requested; Started; Finished], closureComputations["FileFirst.fs"])
\ No newline at end of file
+    Assert.Equal<JobEvent list>([Weakened; Requested; Started; Finished; Weakened; Requested; Started; Finished], closureComputations["FileFirst.fs"])
+    
+[<Fact>]
+let ``TryGetRecentCheckResultsForFile returns None before first call to ParseAndCheckFileInProject`` () =
+    let project = SyntheticProject.Create(
+        sourceFile "First" [])
+    
+    let getSource f = f |> getSourceText project :> ISourceText |> Some |> async.Return
+    
+    ProjectWorkflowBuilder(project, useTransparentCompiler = true) {
+        withProject( fun project checker ->
+                async {
+                    let projectOptions = project.GetProjectOptions(checker)
+                    let projectSnapshot = FSharpProjectSnapshot.FromOptions(projectOptions, DocumentSource.Custom getSource) |> Async.RunSynchronously
+                    let r = checker.TryGetRecentCheckResultsForFile(project.SourceFilePaths[0], projectSnapshot) |> Async.RunSynchronously
```

Comment:
These tests would look much better if we made `TryGetRecentCheckResultsForFile` into a custom operation, which would take the `Id` (`"First"`) of the file and create the snapshot from the context. Then you'd have something like

```fsharp
project.Workflow { 
   tryGetRecentCheckResult "Second" expectNone
   checkFile "Second" expectOk
   tryGetRecentCheckResult "Second" expectOk // or something... 
   updateFile "First" changePublicSurface 
   tryGetRecentCheckResult "Second" expectOk
}